src/openlmis-quantity-unit-input/openlmis-quantity-unit-input.controller.js

```diff
@@ -0,0 +1,136 @@
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+(function() {
+
+    'use strict';
+
+    /**
+     * @ngdoc controller
+     * @name openlmis-quantity-unit-input.controller:QuantityUnitInputController
+     *
+     * @description
+     * Responsible for managing input element for packs and doses.
+     */
+    angular
+        .module('openlmis-quantity-unit-input')
+        .controller('QuantityUnitInputController', controller);
+
+    controller.$inject = ['quantityUnitCalculateService'];
+
+    function controller(quantityUnitCalculateService) {
+
+        var vm = this;
+
+        vm.$onInit = onInit;
+        vm.isQuantityRemainderInDosesDisabled = isQuantityRemainderInDosesDisabled;
+        vm.changeValue = changeValue;
+
+        /**
+         * @ngdoc property
+         * @propertyOf openlmis-quantity-unit-input.controller:QuantityUnitInputController
+         * @name showInDoses
+         * @type {boolean}
+         *
+         * @description
+         * Returns whether the screen is showing quantities in doses.
+         */
+
+        /**
+         * @ngdoc property
+         * @propertyOf openlmis-quantity-unit-input.controller:QuantityUnitInputController
+         * @name onChangeQuantity
+         * @type {Function}
+         *
+         * @description
+         * Method that will be executed on input change with item value as parameter.
+         */
+
+        /**
+         * @ngdoc property
+         * @propertyOf openlmis-quantity-unit-input.controller:QuantityUnitInputController
+         * @name item
+         * @type {Object}
+         *
+         * @description
+         * Model for the inputs. Holds quantity information in doses and packs
+         */
+
+        /**
+         * @ngdoc property
+         * @propertyOf openlmis-quantity-unit-input.controller:QuantityUnitInputController
+         * @name netContent
+         * @type {Number}
+         *
+         * @description
+         * Number of doses per pack for a given object
+         */
+
+        /**
+         * @ngdoc property
+         * @propertyOf openlmis-quantity-unit-input.controller:QuantityUnitInputController
+         * @name inputClass
+         * @type {String}
+         *
+         * @description
+         * Holds information about css classes
+         */
```

Comment:
I would consider moving this documentation to into `src/openlmis-quantity-unit-input/openlmis-quantity-unit-input.component.js`.

Reply:
good point, moved